File: ECE39500/exams/midterm-2/midterm-2/PartA/VCVState.py
from CompilerState import CompilerState
from StateFactory import StateFactory
import sys

class VCVState(CompilerState):
    INT_MIN = -sys.maxint - 1

    def __init__(self, compiler):
        self.compiler = compiler
        return
    
    def comma(self):
        print("Parsing VCVState, expected int, found ,")
        self.compiler.state = StateFactory.getState("ErrorState")

    def value(self, s):
        self.compiler.valueCount += 1
        self.compiler.values[self.compiler.valueCount] = int(s)
        self.compiler.state = StateFactory.getState("CVState")

    def eol(self):
        print(f"Parsing VCVState, expected value, found eol")
        self.compiler.state = StateFactory.getState("ErrorState")

    def eof(self):
        print("Parsing VCVState, expected value, found eof")
        self.compiler.state = StateFactory.getState("ErrorState")
    
    def __str__(self):
        return "VCVState"

File: ECE39500/exams/bak_midterm-2/PartA/VCVState.java
public class VCVState extends CompilerState {

   Compiler compiler = null; 

   public VCVState(Compiler compiler) {
      this.compiler = compiler;
   }

   public void comma( ) { 
      System.out.println("Parsing VCVState, expected int, found ,");
      compiler.state = StateFactory.getState("ErrorState");
   }

   public void value(String s) {
      compiler.values[++compiler.valueCount] = Integer.parseInt(s);
      compiler.state = StateFactory.getState("CVState");
   }

   public void eol( ) { 
      System.out.println("Parsing VCVState, expected value, found eol");
      // XXX compiler.targetState.eol( );
      compiler.state = StateFactory.getState("ErrorState");
   }
      
   public void eof( ) {
      System.out.println("Error in VCVState, expected value, found eof");
      compiler.state = StateFactory.getState("ErrorState");
   }

   public String toString( ) {
      return "VCVState";
   }
}
